### PR TITLE
Create using-foundry-with-the-avalanche-c-chain.md

### PR DESCRIPTION
**Pull request by Julian-dev28**
A smart contract deployment tutorial that uses [Avalanche Smart Contract Quickstart](https://github.com/ava-labs/avalanche-smart-contract-quickstart) and [Foundry](https://book.getfoundry.sh/index.html)

### CODE REVIEW

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    // _setTokenURI(newItemId, tokenURI);
+
+    return newItemId;
+  }
+}
+```
+
+Let's examine this implementation of an NFT as a Game Item. We start by importing to contracts from our node modules. We import Openzeppelin's open source implementation of the ERC721 standard which our NFT contract will inherit from. Our constructor takes the `_name` and `_symbol` arguments for our NFT and passes them on to the constructor of the parent ERC721 implementation. Lastly we implement the `awardItem` function which allows anyone to mint an NFT to a player's wallet address. This function increments the `currentTokenId` and makes use of the `_mint` function of our parent contract.
+
+# Compile & deploy with forge
```
Comment (jpop32) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:68]: What is 'Forge'? A brief introduction might be useful.

Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:68]: Definitely!

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# Using Foundry with the Avalanche C Chain
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:1]:
```suggestion
# Using Foundry with the Avalanche C-Chain
```

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+Clone the Avalanche Smart Contract Quickstart repo and install its dependencies by running:
+
+```zsh
+git clone https://github.com/ava-labs/avalanche-smart-contract-quickstart.git
+cd avalanche-smart-contract-quickstart
+yarn
+```
+
+In order to deploy contracts, you need to have some AVAX. You can use a pre-funded account on a local network or get testnet AVAX from the [Avalanche Faucet](https://faucet.avax.network), which is an easy way to get to play around with Avalanche. After getting comfortable with your code, you can run it on Mainnet after making the necessary changes to your workflow.
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:28]: This article is for Fuji. There is no prefunded account on Fuji.

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 1 above)
Comment (yulin-dong) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:68]: Please add info on how to install Forge which probably should be in the "Getting Started" section.

**Review by yulin-dong — CHANGES_REQUESTED** [review]

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+Since we are deploying to Fuji testnet, our `RPC_URL` export should be:
+
+```zsh
+export RPC_URL=https://api.avax-test.network/ext/bc/C/rpc
+```
+
+Once set, you can deploy your NFT with Forge by running the command below while adding the relevant constructor arguments of the NFT contract:
+
+```zsh
+forge create NFT --rpc-url=$RPC_URL --private-key=$PRIVATE_KEY --constructor-args <NFT-NAME> <NFT-SYMBOL>
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:94]: Where are the value of < NFT-NAME > and < NFT-SYMBOL > from and how are they related to the code above?

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+```zsh
+[⠔] Compiling...
+No files changed, compilation skipped
+Deployer: 0x8db97c7cece249c2b98bdc0226cc4c2a57bf52fc
+Deployed to: 0x52c84043cd9c865236f11d9fc9f56aa003c1f922
+Transaction hash: 0xf35c40dbbdc9e4298698ad1cb9937195e5a5e74e557bab1970a5dfd42a32f533
+```
+
+# Using Cast to Interact with the Smart Contract
+We can call functions on our NFT contract with [Cast](https://book.getfoundry.sh/reference/cast/cast-send.html), Foundry's command-line tool for interacting with smart contracts, sending transactions, and getting chain data. In this scenario, we will mint a Game Item to a player's wallet using the [`awardItem` function](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/0f29cbb6375a1a452579213f688609c880d52c01/contracts/NFT.sol#L17) in our smart contract.
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:110]: Please add info on how to install Cast in the "Getting Started" section.

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+Given that you already set your RPC and private key env variables during deployment, mint an NFT from your contract by running:
+
+```zsh
+cast send --rpc-url=$RPC_URL <NFT-CONTRACT-ADDRESS> "awardItem(address)" <RECIPIENT-ADDRESS> --private-key=$PRIVATE_KEY
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:115]: How to set < NFT-CONTRACT-ADDRESS > and < NFT-CONTRACT-ADDRESS >?

**Review by yulin-dong — CHANGES_REQUESTED** [review]

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 1 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:68]: ok, it is added during foundry installation so I'll word a description like this

prerequisites
- You have installed [foundry](https://book.getfoundry.sh/getting-started/installation.html); This installation includes the `forge` and `cast` binaries used in this walk-through.

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 3 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:28]: roger that ty

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 4 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:94]: Once set, you can [deploy your NFT with Forge](https://book.getfoundry.sh/reference/forge/forge-create.html) by running the command below while adding the values for `_name` and `_symbol`, the relevant [constructor arguments](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/3ad93abf50fba65e3aab68f23382bcace73968be/contracts/NFT.sol#L13) of the NFT contract:

```zsh
forge create NFT --rpc-url=$RPC_URL --private-key=$PRIVATE_KEY --constructor-args GameItem ITM 
```

Hows this?

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 6 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:115]: How does this sound?

Mint an NFT from your contract by replacing `<NFT-CONTRACT-ADDRESS>` with your `Deployed to` address and `<NFT-RECIPIENT-ADDRESS>` with an address of your choice.
_This section assumes you already set your RPC and private key env variables during deployment_

**Comment by Julian-dev28** [comment]
> Blocked by bug in Local Network Workflow
> 
> Followed up with Foundry telegram support to resolve the issue

Fixed 16JUN2022 

Solution: Download Foundry using [`foundryup`](https://github.com/foundry-rs/foundry#installation)

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
+# Using Foundry with the Avalanche C-Chain
+
+## Introduction
+
+This article shows how to deploy and interact with smart contracts using foundry on the [Fuji C-Chain](../../quickstart/fuji-workflow.md).
+
+[Foundry](https://github.com/foundry-rs/foundry) is a smart contract development toolchain written in Rust.
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:7]: toolchain?

**Review by yulin-dong — CHANGES_REQUESTED** [review]

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+Foundry manages your dependencies, compiles your project, runs tests, deploys, and lets you interact with the chain from the command-line.
+
+## Prerequisites
+
+- You have installed [Foundry](https://github.com/foundry-rs/foundry#installation); This installation includes the `forge` and `cast` binaries used in this walk-through
+- You are familiar with [Avalanche Smart Contract Quickstart](https://github.com/ava-labs/avalanche-smart-contract-quickstart)
+- If you plan on running locally, ensure that you have installed and are familiar with [Avalanche Network Runner](../../quickstart/network-runner)
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:15]:
```suggestion
- If you plan on running locally, ensure that you have installed and are familiar with [Avalanche Network Runner](../../quickstart/network-runner).
```

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+This article shows how to deploy and interact with smart contracts using foundry on the [Fuji C-Chain](../../quickstart/fuji-workflow.md).
+
+[Foundry](https://github.com/foundry-rs/foundry) is a smart contract development toolchain written in Rust.
+
+Foundry manages your dependencies, compiles your project, runs tests, deploys, and lets you interact with the chain from the command-line.
+
+## Prerequisites
+
+- You have installed [Foundry](https://github.com/foundry-rs/foundry#installation); This installation includes the `forge` and `cast` binaries used in this walk-through
+- You are familiar with [Avalanche Smart Contract Quickstart](https://github.com/ava-labs/avalanche-smart-contract-quickstart)
```
Comment (yulin-dong) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:14]:
```suggestion
- You are familiar with [Avalanche Smart Contract Quickstart](https://github.com/ava-labs/avalanche-smart-contract-quickstart).
```

**Review by yulin-dong — CHANGES_REQUESTED** [review]

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+This article shows how to deploy and interact with smart contracts using foundry on the [Fuji C-Chain](../../quickstart/fuji-workflow.md).
+
+[Foundry](https://github.com/foundry-rs/foundry) is a smart contract development toolchain written in Rust.
+
+Foundry manages your dependencies, compiles your project, runs tests, deploys, and lets you interact with the chain from the command-line.
+
+## Prerequisites
+
+- You have installed [Foundry](https://github.com/foundry-rs/foundry#installation); This installation includes the `forge` and `cast` binaries used in this walk-through
```
Comment (cgcardona) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:13]: 1. Please make the link be for `installed Foundry` and not just for `Foundry`.

```text
installed [Foundry](https://github.com/foundry-rs/foundry#installation); 
```


Should be 

```text
[installed Foundry](https://github.com/foundry-rs/foundry#installation); 
```

2. Replace the `;` at the end of the sentence with a `.`

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+yarn
+```
+
+In order to deploy contracts, you need to have some AVAX. You can get testnet AVAX from the [Avalanche Faucet](https://faucet.avax.network), which is an easy way to get to play around with Avalanche. After getting comfortable with your code, you can run it on Mainnet after making the necessary changes to your workflow.
+
+## Implement a Game Item NFT
+
+We will use our example ERC721 smart contract, [`NFT.sol`](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/3fbba0ac28f6420e9be5d2635d5f23693f80127a/contracts/NFT.sol) found in `./contracts` of our project.
+
+```ts
```
Comment (cgcardona) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:33]: Can we have this block of text be of type `solidity` ? I'm not sure if that works. If not then `ts` is good.

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+    uint256 newItemId = _tokenIds.current();
+    _mint(player, newItemId);
+    // _setTokenURI(newItemId, tokenURI);
+
+    return newItemId;
+  }
+}
+```
+
+Let's examine this implementation of an NFT as a Game Item. We start by importing to contracts from our node modules. We import Openzeppelin's open source implementation of the ERC721 standard which our NFT contract will inherit from. Our constructor takes the `_name` and `_symbol` arguments for our NFT and passes them on to the constructor of the parent ERC721 implementation. Lastly we implement the `awardItem` function which allows anyone to mint an NFT to a player's wallet address. This function increments the `currentTokenId` and makes use of the `_mint` function of our parent contract.
```
Comment (cgcardona) [inline docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:65]: Link to OpenZeppelin's ERC721 docs here

```text
We import Openzeppelin's open source implementation of the [ERC721 standard](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721)
```

Fix `Openzeppeliln's` to `OpenZeppelin's`

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 11 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:33]: solidity is not yet supported :(

File: docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md (hunk 12 above)
Comment (Julian-dev28) [reply docs/dapps/smart-contracts/using-foundry-with-the-avalanche-c-chain.md:65]: will do

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**